src/encryption/asymmetric/lwe/mod.rs

```diff
@@ -0,0 +1,186 @@
+  let mut rng = rand::thread_rng();
+  let mut sum = PrimeField::<Q>::ZERO;
+
+  for _ in 0..K {
+    if rng.gen::<bool>() {
+      sum += PrimeField::<Q>::ONE;
+    }
+    if rng.gen::<bool>() {
+      sum -= PrimeField::<Q>::ONE;
+    }
+  }
+  sum
+}
+
+impl<const Q: usize, const N: usize, const K: usize> LWE<Q, N, K> {
+  pub fn new() -> LWE<Q, N, K> {
+    let mut rng = rand::thread_rng();
+
+    // Generate random matrix A
+    let mut a = [[PrimeField::<Q>::ZERO; N]; N];
+    for i in 0..N {
+      for j in 0..N {
+        a[i][j] = PrimeField::from(rng.gen_range(0..Q));
+      }
+    }
+
+    // Sample secret vector s with small coefficients
+    let mut s = [PrimeField::<Q>::ZERO; N];
+    for i in 0..N {
+      s[i] = sample_binomial::<Q, K>();
+    }
+
+    // Generate error vector e
+    let mut e = [PrimeField::<Q>::ZERO; N];
+    for i in 0..N {
+      e[i] = sample_binomial::<Q, K>();
+    }
+
+    // Compute b = As + e
+    let mut b = [PrimeField::<Q>::ZERO; N];
+    for i in 0..N {
+      let mut sum = PrimeField::<Q>::ZERO;
+      for j in 0..N {
+        sum += a[i][j] * s[j];
+      }
+      sum += e[i];
+      b[i] = sum;
+    }
+
+    Self { public_key: PublicKey { a, b }, private_key: PrivateKey { s } }
+  }
+}
+
+impl<const Q: usize, const N: usize, const K: usize> AsymmetricEncryption for LWE<Q, N, K> {
+  type Ciphertext = Ciphertext<Q, N>;
+  type Plaintext = bool;
+  type PrivateKey = PrivateKey<Q, N>;
+  type PublicKey = PublicKey<Q, N>;
+
+  fn encrypt(&self, plaintext: &Self::Plaintext) -> Self::Ciphertext {
+    let mut rng = rand::thread_rng();
+
+    // Sample random vector r (binary)
+    let mut r = [PrimeField::<Q>::ZERO; N];
+    for i in 0..N {
+      r[i] = PrimeField::from(rng.gen_range(0..2));
+    }
+
+    // Compute u = A^T r
+    let mut u = [PrimeField::<Q>::ZERO; N];
+    for i in 0..N {
+      for j in 0..N {
+        u[i] += self.public_key.a[j][i] * r[j];
+      }
+    }
+
+    // Compute v = b^T r + ⌊q/2⌋m
+    let mut v = PrimeField::<Q>::ZERO;
+    for i in 0..N {
+      v += self.public_key.b[i] * r[i];
+    }
+
+    if *plaintext {
+      v += PrimeField::from(Q / 2);
+    }
+
+    Ciphertext { u, v }
+  }
+
+  fn decrypt(&self, ct: &Self::Ciphertext) -> Self::Plaintext {
+    // Compute v - s^T u
+    let mut result = ct.v;
+    for i in 0..N {
+      result -= self.private_key.s[i] * ct.u[i];
+    }
+
+    // Get q/2 as a field element
+    let q_half = PrimeField::<Q>::from(Q / 2);
+
+    // For distance to zero, we need min(x, q-x)
+    let dist_to_zero = result.min(-result);
+
+    // For distance to q/2, we need min(|x - q/2|, |q/2 - x|)
+    let dist_to_q_half = if result >= q_half {
+      // If result ≥ q/2, distance is min(result - q/2, q - result + q/2)
+      (result - q_half).min(-result + q_half)
+    } else {
+      // If result < q/2, distance is min(q/2 - result, result + q/2)
+      (q_half - result).min(result + q_half)
+    };
```

Comment:
Is this needed?
I think `(result - q_half) <= (-result + q_half)` for `result >= q_half`, and `(q_half - result) <= (result + q_half)` for `result < q_half`, is always true. I also did a quick check if this is ever **not true** for `test_encryption_decryption`. 

Maybe you could verify this?

Maybe this could be distilled to:
```suggestion
    let dist_to_q_half = (result - q_half).min(q_half - result);
```